src/sentry/release_health/release_monitor/metrics.py

```diff
@@ -0,0 +1,203 @@
+                            ),
+                        ],
+                        granularity=Granularity(3600),
+                        orderby=[
+                            OrderBy(Column("org_id"), Direction.ASC),
+                            OrderBy(Column("project_id"), Direction.ASC),
+                        ],
+                    )
+                    .set_limit(self.CHUNK_SIZE + 1)
+                    .set_offset(offset)
+                )
+                data = raw_snql_query(
+                    query, referrer="release_monitor.fetch_projects_with_recent_sessions"
+                )["data"]
+                count = len(data)
+                more_results = count > self.CHUNK_SIZE
+                offset += self.CHUNK_SIZE
+
+                if more_results:
+                    data = data[:-1]
+
+                for row in data:
+                    aggregated_projects[row["org_id"]].append(row["project_id"])
+
+                if not more_results:
+                    break
+
+            else:
+                logger.error(
+                    "release_monitor.fetch_projects_with_recent_sessions.loop_timeout",
+                    extra={"offset": offset},
+                )
+
+        return aggregated_projects
+
+    def fetch_project_release_health_totals(
+        self, org_id: int, project_ids: Sequence[int]
+    ) -> Totals:
+        start_time = time.time()
+        offset = 0
+        totals: Totals = defaultdict(dict)
+        with metrics.timer("release_monitor.fetch_project_release_health_totals.loop"):
+            while (time.time() - start_time) < self.MAX_SECONDS:
+                # query = QueryDefinition(
+                #     org_id=org_id,
+                #     project_ids=project_ids,
+                #     select=[
+                #         MetricField("sum", "sentry.sessions.session"),
+                #         MetricField(None, "project_id"),
+                #         MetricField(None, "release"),
+                #         MetricField(None, "environment"),
+                #     ],
+                #     start=datetime.utcnow() - timedelta(hours=6),
+                #     end=datetime.utcnow(),
+                #     granularity=Granularity(21600),
+                #     where=[Condition(
+                #         Column("metric_id"),
+                #         Op.EQ,
+                #         indexer.resolve(org_id, SessionMRI.SESSION.value),
+                #     )],
+                #     groupby=[
+                #         Column("project_id"),
+                #         Column(resolve_tag_key(org_id, "release")),
+                #         Column(resolve_tag_key(org_id, "environment")),
+                #     ],
+                #     orderby=MetricsOrderBy(MetricField(None, "project_id"), Direction.ASC),
+                #     limit=self.CHUNK_SIZE + 1,
+                #     offset=offset,
+                # )
```

Comment:
@ahmedetefy I tried to take your advice and use this with `get_series` but had no luck. It wouldn't let me sort by `project_id` since it seemed to expect that I'd be using an aggregate to sort. `MetricsOrderBy` also seems to only allow a single column.

I've managed to get this to work just using `raw_snql_query` for now at least.

Reply:
Yeah, we only allow sorting on a single aggregate for now. Sorting by project_id could be implemented, but sorting by release and environment in a meaningful way is currently impossible because they are integers in snuba.

@wedamija are you sorting by `release_col`, `env_col` for stability? Or do you actually need these strings to come out in alphabetical order?

Reply:
@jjbayer Just for stability, I don't care about alphabetical order. I assume this will work fine since the int representations here won't change within a project.

Reply:
Thinking about it more, I don't need these, they're more of an improvement. We don't sort by these fields in the sessions version. So if that's a blocker I could remove them so that we can convert this to use the abstraction.

I'm also open to just merging as is and converting over once the abstraction is more flexible.

Reply:
But sorting by `project_id` is a requirement, right? Since `get_series` does not support that yet, I would also just merge as is and convert later. Either way I left some comments on the `QueryDefinition` to clarify how it's intended to work.

@ahmedetefy I'll create a jira issue for "order by project_id"